src/main/kotlin/day16/practice problems 04.kt

```
@@ -0,0 +1,103 @@
+        val dataStatus = HttpClient(mockEngine).get("https://jsonplaceholder.typicode.com/todos/$id")
+        // ClientRequestException가 4xx 예외를 못 잡아준다.
+//        try {
+//            data = HttpClient(mockEngine).get("https://jsonplaceholder.typicode.com/todos/$id").bodyAsText()
+//            println("try")
+//        } catch(e: ClientRequestException) {
+//            data = HttpClient(mockEngine).get("https://jsonplaceholder.typicode.com/todos/1").bodyAsText()
+//            println("catch")
+//        }
+        var data: String = ""
+        return if(dataStatus.status == HttpStatusCode.NotFound)
+            throw IllegalArgumentException()
```

detekt (1.23.7)

[warning] A call to the default constructor of an exception was detected. Instead one of the constructor overloads should be called. This allows to provide more meaningful exceptions.

(detekt.exceptions.ThrowingExceptionsWithoutMessageOrCause)